client.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: The vmnet-helper authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+// Apple recommend receive buffer size to be 4 times the size of the send
+// buffer size, but send buffer size is not used to allocate a buffer in
+// datagram sockets, it only limits the maximum packet size.
+// Must be larger than TSO packets size (65550 bytes).
+static const int SNDBUF_SIZE = 65 * 1024;
+
+// The receive buffer size determine how many packets can be queued by the
+// peer. Using bigger receive buffer size make ENOBUFS error less likey for the
+// peer and improves throughput.
+static const int RCVBUF_SIZE = 4 * 1024 * 1024;
+
+static void set_socket_buffers(int fd)
+{
+    // Setting socket buffer size is a performance optimization - don't fail on
+    // errors.
+    if (setsockopt(fd, SOL_SOCKET, SO_SNDBUF, &SNDBUF_SIZE, sizeof(SNDBUF_SIZE)) < 0) {
+        perror("setsockopt");
+    }
+
+    if (setsockopt(fd, SOL_SOCKET, SO_RCVBUF, &RCVBUF_SIZE, sizeof(RCVBUF_SIZE)) < 0) {
+        perror("setsockopt");
+    }
+}
+
+int main(int argc, char **argv)
+{
+    if (argc < 2) {
+        fprintf(stderr, "Usage: vmnet-client command ...\n");
+        exit(EXIT_FAILURE);
+    }
+
```

Comment:
I think it is pretty safe to assume and if you manage to run the helper without one of the standard descriptor you deserve what you get. However we can inherit more descriptor from the caller which we don't need and we should not inherit, so the socketpair may return other values like 4,5 instead of 3,4.

We can avoid this by closing descriptors 3 and 4 before creating the socketpair.